pkg/users/accounts-list.js

```diff
@@ -18,72 +18,277 @@
+        loginText = _("Logged in");
+        loginSortKey = "logged in";
+    } else if (!account.lastLogin) {
+        loginText = _("Never logged in");
+        loginSortKey = "never";
+    } else {
+        loginSortKey = new Date(account.lastLogin);
+        loginText = timeformat.dateTime(loginSortKey);
     }
 
-    return (
-        <Card onClick={click} onKeyPress={click} isSelectable isCompact>
-            <CardBody className="cockpit-account">
-                <UserIcon className="cockpit-account-pic" />
-                <div className="cockpit-account-real-name">{account.gecos.split(',')[0]}</div>
-                <div className="cockpit-account-user-name">
+    const columns = [
+        {
+            title: (
+                <span>
                     <a href={"#/" + account.name}>{account.name}</a>
-                    {current && <Badge className="cockpit-account-badge">{_("Your account")}</Badge>}
-                </div>
-            </CardBody>
-        </Card>
-    );
-}
+                    {current && <Badge className="pf-u-ml-lg" id="current-account-badge">{_("Your account")}</Badge>}
+                </span>
+            ),
+            sortKey: account.name,
+            props: { width: 25, },
+        },
+        {
+            title: account.gecos.split(',')[0],
+            props: { width: 20, },
+        },
+        {
+            title: account.uid.toString(),
+            props: { width: 10, },
+        },
+        {
+            title: loginText,
+            sortKey: loginSortKey,
+            props: { width: 25, },
+        },
+        {
+            title: (
+                <Flex spaceItems={{ default: 'spaceItemsSm' }}>
+                    {userGroupLabels}
+                </Flex>
+            ),
+            props: { width: 20, className: "th-align-right" },
+        },
+        {
+            title: <UserActions account={account} />,
+        },
+    ];
 
-export function AccountsList({ accounts, current_user }) {
-    const filtered_accounts = accounts.filter(function(account) {
-        return !((account.uid < 1000 && account.uid !== 0) ||
-                 account.shell.match(/^(\/usr)?\/sbin\/nologin/) ||
-                 account.shell === '/bin/false');
-    });
+    return { columns };
+};
+
+const mapGroupsToAccount = (accounts, groups) => {
+    return accounts.map(account => {
+        const accountGroups = [];
+        groups.forEach(group => {
+            if (group.userlist.find(accountName => accountName === account.name))
+                accountGroups.push(group.name);
+        });
+        account.groups = accountGroups;
 
-    filtered_accounts.sort(function (a, b) {
-        if (current_user === a.name) return -1;
-        else if (current_user === b.name) return 1;
-        else if (!a.gecos) return -1;
-        else if (!b.gecos) return 1;
-        else return a.gecos.localeCompare(b.gecos);
+        return account;
     });
+};
+
+export class AccountsList extends React.Component {
```

Comment:
This would be nicer as a functional component. Generally I like to avoid using class components when the state updates are so simple like here. Just for the sake of avoiding the extra boilerplate code.
Your choice here obviously.

Reply:
Oops. I forgot to change this. Yes, this was the plan.